p2p/conn/secret_connection.go

```diff
@@ -313,22 +313,19 @@ func shareAuthSignature(sc *SecretConnection, pubKey crypto.PubKey, signature cr
 
 // sha256
 func hash32(input []byte) (res *[32]byte) {
-	hasher := sha256.New()
-	hasher.Write(input) // nolint: errcheck, gas
-	resSlice := hasher.Sum(nil)
+	hash := sha256.New
+	hkdf := hkdf.New(hash, input, nil, nil)
 	res = new([32]byte)
-	copy(res[:], resSlice)
-	return
+	io.ReadFull(hkdf, res[:])
+	return res
 }
 
-// We only fill in the first 20 bytes with ripemd160
 func hash24(input []byte) (res *[24]byte) {
-	hasher := ripemd160.New()
-	hasher.Write(input) // nolint: errcheck, gas
-	resSlice := hasher.Sum(nil)
+	hash := sha256.New
+	hkdf := hkdf.New(hash, input, nil, nil)
```

Comment:
don't we need to pass the key in here as well to actually utilize it as HKDF ? otherwise we might as well just SHA256(input) ? 

Also I think Tony's recommending we use something like "TENDERMINT_SECRETCONNECTION_KDF" for the `info` param

Reply:
Adding a salt like that seems like a good practice.

The KDF is taking the constructor as an argument not the hash object.